bindings/ruby/lib/typelib/gccxml.rb

```diff
@@ -830,13 +830,24 @@ def self.preprocess(files, kind, options)
                     io.puts "#include <#{path}>"
                 end
                 io.flush
-                result = IO.popen(["gccxml", "--preprocess", *includes, *defines, *gccxml_default_options, io.path]) do |gccxml_io|
-                    gccxml_io.read
-                end
-                if !$?.success?
-                    raise ArgumentError, "failed to preprocess #{files.join(" ")}"
+                Tempfile.open('gccxml_err') do |err|
+                    command = ["gccxml", "--preprocess", *includes, *defines, *gccxml_default_options, io.path]
+                    result = IO.popen(command, :err => err.path) do |gccxml_io|
+                        gccxml_io.read
+                    end
+                    if !$?.success?
+                        Typelib.debug do
+                            FileUtils.copy_file(io.path, "/tmp/gcc-debug")
+                            "Command was \"#{command.join(" ")}\".
```

Comment:
This breaaks logger formatting. If you want multiple lines, do

~~~
Typelib.debug do
   Typelib.debug "line0"
   Typelib.debug "line1"
   break
end
~~~